File: q9.js
const fs = require('fs');
let words = fs.readFileSync('wordle.txt', 'utf8').split('\n');
function exercise9(green, yellow, gray) {
    let wordle = [];
    for (let i of words) {
        let flag = [...gray].some(j => i.includes(j));
        flag = flag || Object.entries(green).some(([pos, letter]) => i[pos] !== letter);
        flag = flag || Object.entries(yellow).some(([letter, positions]) => [...positions].some(pos => i[pos] === letter) || !i.includes(letter));
        if (!flag) {
            wordle.push(i);
        }
    }
    return wordle.length;
}
const green_2 = {2:'a'}
const yellow_2 = {'a':new Set([3]),'i':new Set([2]),'l':new Set([3,4]),'r':new Set([1])}
const gray_2 = new Set(['e', 't', 'u', 'o', 'p', 'g', 'h', 'c', 'm', 's'])

console.log(exercise9(green_2, yellow_2, gray_2));

File: q9.py
from re import search
text = "C:/Users/vigne/Downloads/wordle.txt"
f= open(text)
f = f.readlines()
f = [x.strip('\n') for x in f]
wordle = []
def exercise9(green,yellow,gray):
    flag = 0
    for i in f:
        for j in gray:
            if search(j,i):
                flag = 1
                #break
                
    #for i in f:
        for j in green:
            
            if(i[j]!=green[j]):
                flag = 1
                #break
    #for i in f:
        for j in yellow:
            for k in yellow[j]:
                #print(j)
                if(i[k]==j):
                    flag=1
                elif j not in i:
                    flag=1  
        if flag==0:
            wordle.append(i)
        flag = 0
            
            
    return wordle
                
                
        
    
    
print(exercise9(green = {1:'i',3:'c'}, yellow = {'e':{3}},gray = {'r','a','s','d','f'}))

# green = {'i':1,3:'c'}
# for i in green:
#     print(green[i])